### ISSUE TITLE
Validation fails when claim has boolean false as value

### ISSUE THREAD

**Issue by stormcow** [authlib/joserfc#23]
Hello, I have noticed that when passing a claim that has a boolean false as value, the validation fails with Invalid claim error, e.g.:

```
registry = jwt.JWTClaimsRegistry(bool_claim={"essential": True})
registry.validate({"bool_claim": False})
```

```
Traceback (most recent call last):
  File "/home/schemer/test/test/main.py", line 23, in <module>
    registry.validate({"bool_claim": False})
  File "/home/schemer/test/.venv/lib/python3.12/site-packages/joserfc/rfc7519/registry.py", line 52, in validate
    self.check_value(key, value)
  File "/home/schemer/test/.venv/lib/python3.12/site-packages/joserfc/rfc7519/registry.py", line 31, in check_value
    raise InvalidClaimError(claim_name)
joserfc.errors.InvalidClaimError: invalid_claim: Invalid claim: "bool_claim"
```

But the same registry validates correctly when the claim has a boolean true:

```
registry = jwt.JWTClaimsRegistry(bool_claim={"essential": True})
registry.validate({"bool_claim": True})
```

I went through the code, and found that the issue is getting raised here:
https://github.com/authlib/joserfc/blob/18ae2e263365569a11cc2c55049ddf7210ff416d/src/joserfc/rfc7519/registry.py#L30

Passing the "allow_blank" option does fix this, but I don't want to have a blank claim and I want to always ensure that it has a value.

**Comment by stormcow** [comment]
The exception is raised on all falsy values of a claim without the "allow_blank" option. So if I were for example to pass this claim:
```
registry.validate({"int_claim": 0})
```
i will still get an exception, even though it isn't null nor is it empty.
```
Traceback (most recent call last):
  File "/home/schemer/test/test/main.py", line 28, in <module>
    registry.validate({"int_claim": 0})
  File "/home/schemer/test/.venv/lib/python3.12/site-packages/joserfc/rfc7519/registry.py", line 52, in validate
    self.check_value(key, value)
  File "/home/schemer/test/.venv/lib/python3.12/site-packages/joserfc/rfc7519/registry.py", line 31, in check_value
    raise InvalidClaimError(claim_name)
joserfc.errors.InvalidClaimError: invalid_claim: Invalid claim: "int_claim"
```

Is this behaviour intended? I'm very confused :/

**Comment by lepture** [comment]
Released 0.12.0
